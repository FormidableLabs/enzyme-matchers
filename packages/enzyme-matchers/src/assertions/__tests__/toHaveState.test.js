const toHaveState = require('../toHaveState');

class Fixture extends React.Component {
  constructor() {
    super();
    this.state = {
      foo: false,
      array: [1, 2, 3],
      object: { foo: 'bar' },
    };
  }

  render() {
    return <div />;
  }
}

describe('toHaveState', () => {
  [shallow, mount].forEach(builder => {
    describe(builder.name, () => {
      function build() {
        const wrapper = builder(<Fixture />);
        const truthyResults = toHaveState(wrapper, 'array', [1, 2, 3]);
        const falsyResults = toHaveState(wrapper, 'array', [4, 5, 6]);

        return {
          truthyResults,
          falsyResults,
        };
      }

      it('returns the pass flag properly', () => {
        const { truthyResults, falsyResults } = build();

        expect(truthyResults.pass).toBe(true);
        expect(falsyResults.pass).toBe(false);
      });

      it(`returns the message with the proper pass verbage (${builder.name})`, () => {
        const { truthyResults } = build();
        expect(truthyResults.message).toMatchSnapshot();
      });

      it(`returns the message with the proper fail verbage (${builder.name})`, () => {
        const { truthyResults } = build();
        expect(truthyResults.negatedMessage).toMatchSnapshot();
      });

      it(`provides contextual information for the message (${builder.name})`, () => {
        const { truthyResults } = build();
        expect(truthyResults.contextualInformation).toMatchSnapshot();
      });

      it('can validate falsy values', () => {
        const wrapper = shallow(<Fixture />);
        const { pass } = toHaveState(wrapper, 'foo', false);
        const { pass: fail } = toHaveState(wrapper, 'foo', true);

        expect(pass).toBe(true);
        expect(fail).toBe(false);
      });

      it('can validate arrays', () => {
        const wrapper = shallow(<Fixture />);
        const { pass } = toHaveState(wrapper, 'array', [1, 2, 3]);
        const { pass: fail } = toHaveState(wrapper, 'array', [4, 5, 6]);

        expect(pass).toBe(true);
        expect(fail).toBe(false);
      });

      it('can validate objects', () => {
        const wrapper = shallow(<Fixture />);
        const { pass } = toHaveState(wrapper, 'object', { foo: 'bar' });
        const { pass: fail } = toHaveState(wrapper, 'object', { foo: 'NOPE' });

        expect(pass).toBe(true);
        expect(fail).toBe(false);
      });

      it('returns the pass flag properly', () => {
        const wrapper = mount(<Fixture />);
        const truthyResults = toHaveState(wrapper, 'foo', false);
        const falsyResults = toHaveState(wrapper, 'foo', true);

        expect(truthyResults.pass).toBe(true);
        expect(falsyResults.pass).toBe(false);
      });

      it('works without a prop value', () => {
        const wrapper = builder(<Fixture />);
        const truthy = toHaveState(wrapper, 'foo');

        expect(truthy.pass).toBe(true);
      });

      it('works with undefined value', () => {
        const wrapper = builder(<Fixture />);
        const falsy = toHaveState(wrapper, 'foo', undefined);

        expect(falsy.pass).toBe(false);
      });

      it('should support object arguments', () => {
        const wrapper = builder(<Fixture />);

        const passResult = toHaveState(wrapper, {
          foo: false,
          array: [1, 2, 3],
        });
        const failResult = toHaveState(wrapper, {
          foo: true,
          array: [1, 2, 3],
        });

        expect(passResult.pass).toBe(true);
        expect(failResult.pass).toBe(false);
      });
    });
  });
});
