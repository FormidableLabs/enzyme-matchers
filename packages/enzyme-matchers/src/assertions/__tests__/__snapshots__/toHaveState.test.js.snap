// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toHaveState mount provides "any value" contextual information for the message (mount) 1`] = `
Object {
  "actual": "Actual state: [33marray[39m[90m:[39m [33m[90m[90m[[90m
  [34m1[90m
  [34m2[90m
  [34m3[90m
[90m][90m[33m[39m",
}
`;

exports[`toHaveState mount provides contextual information for the message (mount) 1`] = `
Object {
  "actual": "Actual state: [33marray[39m[90m:[39m [33m[90m[90m[[90m
  [34m1[90m
  [34m2[90m
  [34m3[90m
[90m][90m[33m[39m",
  "expected": "Expected state: [33marray[39m[90m:[39m [33m[90m[90m[[90m
  [34m1[90m
  [34m2[90m
  [34m3[90m
[90m][90m[33m[39m",
}
`;

exports[`toHaveState mount returns the "any value" message with the proper fail verbage (mount) 1`] = `"Expected the state for <Fixture> to not contain a value for key \\"array\\", but it did."`;

exports[`toHaveState mount returns the "any value" message with the proper pass verbage (mount) 1`] = `"Expected the state for <Fixture> to contain a value for key \\"array\\", but it did not."`;

exports[`toHaveState mount returns the message with the proper fail verbage (mount) 1`] = `"Expected the state for <Fixture> to not match for props \\"\\", but it did."`;

exports[`toHaveState mount returns the message with the proper pass verbage (mount) 1`] = `"Expected the state for <Fixture> to match for props \\"\\", but it did not."`;

exports[`toHaveState shallow provides "any value" contextual information for the message (shallow) 1`] = `
Object {
  "actual": "Actual state: [33marray[39m[90m:[39m [33m[90m[90m[[90m
  [34m1[90m
  [34m2[90m
  [34m3[90m
[90m][90m[33m[39m",
}
`;

exports[`toHaveState shallow provides contextual information for the message (shallow) 1`] = `
Object {
  "actual": "Actual state: [33marray[39m[90m:[39m [33m[90m[90m[[90m
  [34m1[90m
  [34m2[90m
  [34m3[90m
[90m][90m[33m[39m",
  "expected": "Expected state: [33marray[39m[90m:[39m [33m[90m[90m[[90m
  [34m1[90m
  [34m2[90m
  [34m3[90m
[90m][90m[33m[39m",
}
`;

exports[`toHaveState shallow returns the "any value" message with the proper fail verbage (shallow) 1`] = `"Expected the state for <div> to not contain a value for key \\"array\\", but it did."`;

exports[`toHaveState shallow returns the "any value" message with the proper pass verbage (shallow) 1`] = `"Expected the state for <div> to contain a value for key \\"array\\", but it did not."`;

exports[`toHaveState shallow returns the message with the proper fail verbage (shallow) 1`] = `"Expected the state for <div> to not match for props \\"\\", but it did."`;

exports[`toHaveState shallow returns the message with the proper pass verbage (shallow) 1`] = `"Expected the state for <div> to match for props \\"\\", but it did not."`;
